docs/templates/tour.md

```diff
@@ -0,0 +1,370 @@
+the workspace again.
+
+We do this in 2 parts:
+
+- Our `docker_volume` resource uses the `lifecycle` block with
+  `ignore_changes = all` argument to prevent accidental deletions.
+- To prevent Terraform from destroying persistent Docker volumes in
+  case of a workspace name change, we use an immutable parameter, like
+  `data.coder_workspace.me.id`.
+
+You'll see later that we make sure that our Docker container is
+ephemeral with the Terraform
+[count](https://developer.hashicorp.com/terraform/language/meta-arguments/count)
+meta-argument.
+
+```hcl
+resource "docker_volume" "home_volume" {
+  name = "coder-${data.coder_workspace.me.id}-home"
+  # Protect the volume from being deleted due to changes in attributes.
+  lifecycle {
+    ignore_changes = all
+  }
+}
+```
+
+For details, see [Resource persistence](./resource-persistence.md).
+
+## 6. Set up the Docker container
+
+To set up our Docker container, our template has a `docker_image`
+resource that uses `build/Dockerfile`, which we created earlier.
+
+```hcl
+resource "docker_image" "main" {
+  name = "coder-${data.coder_workspace.me.id}"
+  build {
+    context = "./build"
+    build_args = {
+      USER = local.username
+    }
+  }
+  triggers = {
+    dir_sha1 = sha1(join("", [for f in fileset(path.module, "build/*") : filesha1(f)]))
+  }
+}
+```
+
+Our `docker_container` resource uses the `coder_workspace`
+`start_count` to start and stop the Docker container:
+
+```hcl
+resource "docker_container" "workspace" {
+  count = data.coder_workspace.me.start_count
+  image = docker_image.main.name
+  # Uses lower() to avoid Docker restriction on container names.
+  name = "coder-${data.coder_workspace.me.owner}-${lower(data.coder_workspace.me.name)}"
+  # Hostname makes the shell more user friendly: coder@my-workspace:~$
+  hostname = data.coder_workspace.me.name
+  # Use the docker gateway if the access URL is 127.0.0.1
+  entrypoint = ["sh", "-c", replace(coder_agent.main.init_script, "/localhost|127\\.0\\.0\\.1/", "host.docker.internal")]
+  env = [
+    "CODER_AGENT_TOKEN=${coder_agent.main.token}",
+  ]
+  host {
+    host = "host.docker.internal"
+    ip   = "host-gateway"
+  }
+  volumes {
+    container_path = "/home/${local.username}"
+    volume_name    = docker_volume.home_volume.name
+    read_only      = false
+  }
+}
+```
+
+## 7. Create the template in Coder
+
+We've created the files for our template. Now we can add them to our
+Coder deployment.
+
+We can do this with the Coder CLI or the Coder dashboard. For this
+tour, we'll use the dashboard.
+
+First, we need to package up our template in a tar file:
+
+```console
+$ tar cvf ../template-tour .
+./
+./build/
+./build/Dockerfile
+./main.tf
+$ ls ..
+ coder   template-tour   template-tour.tar
+```
+
+In your web browser, log in to your Coder dashboard, select
+***Templates**, then **Create template**.
+
+Upload the `template-tour.tar` file, the scroll down to select
```

Comment:
It'd probably be better to recommend `coder templates push` which will push the template to Coder from the CLI. The tar upload can be a bit clunky. Thoughts @matifali?

Reply:
Yes. We should recommend using that `template push`

Reply:
@bpmct @matifali It would be `coder templates create` for the first commit. Correct?

Reply:
Yes.

Reply:
Thanks.